stl/inc/xmemory

```diff
@@ -190,98 +190,93 @@ inline void _Adjust_manually_vector_aligned(void*& _Ptr, size_t& _Bytes) {
-    // deallocate storage allocated by _Allocate when __cpp_aligned_new && _Align > __STDCPP_DEFAULT_NEW_ALIGNMENT__
-#if _HAS_CXX20 // TRANSITION, GH-1532
-    if (_STD is_constant_evaluated()) {
-        ::operator delete(_Ptr);
+            return _Traits::_Allocate_aligned(_Bytes, _Passed_align);
+        }
     } else
-#endif // _HAS_CXX20
+#endif // defined(__cpp_aligned_new)
     {
-        size_t _Passed_align = _Align;
-#if defined(_M_IX86) || defined(_M_X64)
-        if (_Bytes >= _Big_allocation_threshold) { // boost the alignment of big allocations to help autovectorization
-            _Passed_align = (_STD max)(_Align, _Big_allocation_alignment);
-        }
-#endif // defined(_M_IX86) || defined(_M_X64)
-        ::operator delete(_Ptr, _Bytes, align_val_t{_Passed_align});
-    }
-}
-
-#define _HAS_ALIGNED_NEW 1
-#else // ^^^ defined(__cpp_aligned_new) / !defined(__cpp_aligned_new) vvv
-#define _HAS_ALIGNED_NEW 0
-#endif // ^^^ !defined(__cpp_aligned_new) ^^^
-
-template <size_t _Align, class _Traits = _Default_allocate_traits,
-    enable_if_t<(!_HAS_ALIGNED_NEW || _Align <= __STDCPP_DEFAULT_NEW_ALIGNMENT__), int> = 0>
-__declspec(allocator) _CONSTEXPR20 void* _Allocate(const size_t _Bytes) {
-    // allocate _Bytes when !_HAS_ALIGNED_NEW || _Align <= __STDCPP_DEFAULT_NEW_ALIGNMENT__
 #if defined(_M_IX86) || defined(_M_X64)
 #if _HAS_CXX20 // TRANSITION, GH-1532
-    if (!_STD is_constant_evaluated())
+        if (!_STD is_constant_evaluated())
 #endif // _HAS_CXX20
-    {
-        if (_Bytes >= _Big_allocation_threshold) { // boost the alignment of big allocations to help autovectorization
-            return _Allocate_manually_vector_aligned<_Traits>(_Bytes);
+        {
+            if (_Bytes >= _Big_allocation_threshold) {
+                // boost the alignment of big allocations to help autovectorization
+                return _Allocate_manually_vector_aligned<_Traits>(_Bytes);
+            }
         }
-    }
 #endif // defined(_M_IX86) || defined(_M_X64)
 
-    if (_Bytes != 0) {
-        return _Traits::_Allocate(_Bytes);
-    }
+        if (_Bytes != 0) {
+            return _Traits::_Allocate(_Bytes);
+        }
 
-    return nullptr;
+        return nullptr;
+    }
 }
 
-template <size_t _Align, enable_if_t<(!_HAS_ALIGNED_NEW || _Align <= __STDCPP_DEFAULT_NEW_ALIGNMENT__), int> = 0>
+template <size_t _Align>
 _CONSTEXPR20 void _Deallocate(void* _Ptr, size_t _Bytes) noexcept {
-    // deallocate storage allocated by _Allocate when !_HAS_ALIGNED_NEW || _Align <= __STDCPP_DEFAULT_NEW_ALIGNMENT__
+    // deallocate storage allocated by _Allocate
+#ifdef __cpp_aligned_new
+    if constexpr (_Align > __STDCPP_DEFAULT_NEW_ALIGNMENT__) {
 #if _HAS_CXX20 // TRANSITION, GH-1532
-    if (_STD is_constant_evaluated()) {
-        ::operator delete(_Ptr);
-    } else
+        if (_STD is_constant_evaluated()) {
+            ::operator delete(_Ptr);
+        } else

```

Comment:
The diff looks horrible, but I was able to validate that the transformation is very clean - thanks! :heart_eyes_cat:

Now we can go even further. There's significant duplication regarding handling `is_constant_evaluated` and other conditions, which we can extract with some care. I'll push a series of commits.